src/Shared/RangeHelper/RangeHelper.cs

```diff
@@ -82,14 +84,14 @@ internal static class RangeHelper
             }
 
             // Normalize the ranges
-            var range = NormalizeRange(ranges.SingleOrDefault(), length);
+            var range = NormalizeRange(ranges.First(), length);
```

Comment:
We guarantee there is at least one element in `ranges`. Do we need to assert this has exactly one item? Using `Single` seems does not provide a useful error message.